src/socket/unix_server.cr

```diff
@@ -29,12 +29,12 @@ class UNIXServer < UNIXSocket
     super(Family::UNIX, type)
 
     bind(UNIXAddress.new(path)) do |error|
-      close
+      close(delete: false)
       raise error
     end
 
     listen(backlog) do |error|
-      close
+      close(delete: false)
```

Comment:
Is there a possibility that the `bind` call above succeeds and creates the file, but `listen` fails? I think in this case the file should be deleted, right?